### ISSUE TITLE
Optimal GPU for testing (CUDA out of memory)

### ISSUE THREAD

**Issue by Gegam** [microsoft/Bringing-Old-Photos-Back-to-Life#103]
I, like many, faced the problem of lack of GPU memory. 

`CUDA out of memory. Tried to allocate...`

I am using RTX3080 with 10gb of memory. Unfortunately, I was not able to figure out the pattern in using the correct image resolution. For example, I process several images in parallel, but an error can pop up on a random image of a very small resolution. Also, I did not understand the reason for the skipping of some images when processing on an AMD Ryzen 9 3900X 12x3.8GHz CPU with 32GB of RAM. Even when processing one image at 1000 x 1000 resolution, this error may be thrown

I also tried to clear the GPU memory after each iteration, but this did not give any result in optimization

`if hasattr(torch.cuda, 'empty_cache'):
	torch.cuda.empty_cache()
`

I have a few questions:
1. Has anyone tried to process images on multi-GPU machines or on GPUs with more memory than the RTX3080? Did you manage to handle large images? If so, what resolution?
2. Did someone manage to establish a pattern for a suitable number and resolution of images?
3. Are there plans to optimize GPU memory usage in the future?

Thanks in advance

**Comment by RhamaDaragaaz** [comment]
I had the same issue and ran into a lot of threads about it, from what I understand pytorch is allowed to allocate (smart allocate) basically all available vram before the process is allowed the try and allocate space for the image to process, which is often why no matter how hard you try you get the out of memory errors with varying sizes of remaining space.

This has been a hot topic for ages in pytorch, but as of less than a month ago a commit was made on the topic that allows setting a fractional total of memory that pytorch is allowed to use.

I would imagine that implementing this here would solve some issues, as even in my 3090 I'm getting the out of memory errors for reasonably sized images.

Here's the commit if you'd like to try implementing a solution:
[Pytorch Memory Limit Commit](https://github.com/pytorch/pytorch/commit/47aa2536328afc51876b2e04384c0cfe71ee1f06)

**Comment by franva** [comment]
hi @RhamaDaragaaz I checked the commit you provided and found the commit has been merged 39 days ago.
How could I confirm this commit is released to which version of PyTorch?

Thanks for your help.

**Comment by SikandAlex** [comment]
This is so frustrating. Can the authors just recommend a maximum image resolution for CoLab? This is an amazing model and it's so frustrating in academia when projects seem to become abandoned. I have an RTX 2080 Ti with 11 GB of VRAM but I believe that that is less or equal to what is provided by CoLab GPUs. @RhamaDaragaaz what resolution can you max at for your 24 GB 3090?

**Comment by zhangmozhe** [comment]
Duplicate of https://github.com/microsoft/Bringing-Old-Photos-Back-to-Life/issues/19#issue-711976416
